integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -239,16 +239,26 @@ private int getattrInternal(String path, FileStat stat) {
       } else {
         status = mFileSystem.getStatus(uri);
       }
+      long size = status.getLength();
```

Comment:
The purpose to get the size here is to make sure the size is gotten at the beginning? Size could be changed during the gap between line 244 and 259?

Reply:
If it could be changed, why not return the value after it is done?

Reply:
Does this change mean return the size even if the file is not finished yet?

Reply:
Alluxio will not update the file length until the file is completed. So when the file is being written, Alluxio Master only have the file metadata showing the file length is 0.
so at line 242, the file length is zero.

This is fine in many cases but also may cause some applications to error out.
e.g. Some applications calls create() - write() - getattr() - write().
If the getattr() between the two writes show written size 0, the application may error out

Reply:
we block waiting for file to be completed when the file is not written by this FUSE.
we cannot block waiting for file is completed when the file is being written by the same FUSE, then the second write() will never be called.